devnulli/EvlWatcher
# final testing of v 2.0

# This is the final pull request before we create the release

## RELEASE NOTES

- is now running on .NetFramework 4.7.2, which is the default framework for Windows servers (2019, ver1809, ver1803) -> see https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/versions-and-dependencies
- is now using a better software architecture pattern, making it easier to change and adapt in the future
- is now using a different configuration file structure
- should fix the ram usage problem of the previous versions
- should fix issues with not banning Event 4625
- fixes a security issue on the named pipe
- the console was revamped

## How to test the beta

The current beta is found [here](https://github.com/devnulli/EvlWatcher/blob/v2-beta2/Versions/v2/EvlWatcher-v2.0%20beta-setup.exe)

- Install EvlWatcher
	- there should be no need to uninstall previous versions, it will be replaced. if that doesn't work, please report that in the pull request

- once installed, check if it runs stable, leave it running for a few days, on a machine with rdp open and exposed to the web. 

- Evlwatcher should still recognize IPs out the event log, according to config.xml
	- if there are any adjustments to be made to these rules, also let me know, please.

- Try out the console. During that pull request, (going until nov 2020) some features will still be added during the next days, and new versions made, including
	- Live Output from the service will be displayed
	- Task configuration and rule tester

**- Stay tuned on that pull request to find out when new versions have been put up.**

Report issues normally on this pull request thread or as an issue




Thanks to all. Especially to shimuldn and NukeDev for helping me out with testing and developing that.

Mike.
